### PR TITLE
Remove q in favor of bluebird

### PR DESCRIPTION
**Pull request by paolobueno**
We were already using bluebird for the unit tests, let's use it for the rest of the lib, and do some cleanup!

Also defferreds are ugly.

:bird:
ping @JameelB @witmicko @nialldonnellyfh

### CODE REVIEW

**Comment by witmicko** [comment]
👍  lgtm

File: lib/topics/index.js [hunk 1]
```diff
@@ -29,7 +29,7 @@ Topics.prototype.getTopic = function(topicName, prefix) {
 
 function wrapInMediatorPromise(self, method, fn) {
```
Comment (JameelB) [inline lib/topics/index.js]: Would it be possible if you could please add comments to this function?

Comment (paolobueno) [inline lib/topics/index.js]: Definitely going to do this as part of a new issue+PR, along with the bug fixes :slightly_smiling_face:

**Comment by paolobueno** [comment]
Merging this to development branch

**Pull request merged**